homework/index.js

```diff
@@ -23,24 +23,147 @@
       const value = options[key];
       if (key === 'text') {
         elem.textContent = value;
+      } else if (key === 'html') {
+        elem.innerHTML = value;
       } else {
         elem.setAttribute(key, value);
       }
     });
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const renderError = error => {
+    const wrapper = document.getElementById('wrapper');
+    wrapper.innerHTML = '';
+    createAndAppend('div', wrapper, {
+      text: error.message,
+      class: 'alert',
+    });
+  };
+
+  const addRepoInfoRow = (parent, label, content) => {
```

Comment:
👍 for doing this in a separate function.

Reply:
Thank you 🙏